core/core.go

```diff
@@ -431,109 +433,27 @@ func runBazel(bazel string, args []string) (int, error) {
 	return 0, nil
 }
 
-type label struct {
-	Name string `json:"name"`
-}
-
-type issue struct {
-	Title  string  `json:"title"`
-	URL    string  `json:"html_url"`
-	Labels []label `json:"labels"`
-}
-
-// IssueList is visible for testing
-type IssueList struct {
-	Items []issue `json:"items"`
-}
-
-// FlagDetails represents an incompatible flag that should be flipped later. It's currently exported for testing purposes.
-type FlagDetails struct {
-	Name          string
-	ReleaseToFlip string
-	IssueURL      string
-}
-
-func (f *FlagDetails) String() string {
-	return fmt.Sprintf("%s (Bazel %s: %s)", f.Name, f.ReleaseToFlip, f.IssueURL)
-}
-
-func getIncompatibleFlags(bazeliskHome, resolvedBazelVersion string) (map[string]*FlagDetails, error) {
-	// GitHub labels use only major and minor version, we ignore the patch number (and any other suffix).
-	re := regexp.MustCompile(`^\d+\.\d+`)
-	version := re.FindString(resolvedBazelVersion)
-	if len(version) == 0 {
-		return nil, fmt.Errorf("invalid version %v", resolvedBazelVersion)
-	}
-	url := "https://api.github.com/search/issues?per_page=100&q=repo:bazelbuild/bazel+label:migration-" + version
-	
-	var issueList IssueList
-
-	merger := func(chunks [][]byte) ([]byte, error) {
-		for _, chunk := range chunks {
-			current, err := ParseIssues(chunk)
-			if err != nil {
-				return nil, err
-			}
-			issueList.Items = append(issueList.Items, current.Items...)
+// getIncompatibleFlags returns all implemented incompatible flags in alphabetical order.
+// The emphasis is on "all"; it does not only return the ones implemented for the actual command.
+// However, the previous implementation didn't do this either, since we most likely run "bazel --migrate build" anyway.
+func getIncompatibleFlags(bazelPath string) ([]string, error) {
+	out := strings.Builder{}
+	if _, err := runBazel(bazelPath, []string{"help", "completion"}, &out); err != nil {
+		return nil, fmt.Errorf("unable to determine incompatible flags with binary %s: %v", bazelPath, err)
+	}
```

Comment:
I played around a bit with our `bazel help` command and I think you can get the flags applicable for a command in relatively easily parsable format using this:

```
bazel help $COMMAND --short | fgrep -- '--[no]incompatible_'
```

There are already a few --incompatible_ flags that do not apply to `build`, but only to e.g. `query`:

```
--incompatible_display_source_file_location
--incompatible_prefer_unordered_output
```

Or to `aquery`:
```
--incompatible_proto_output_v2
```

Or those that are startup options:
```
--incompatible_enable_execution_transition
```